analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Constraints/NumberConstraint.cs

```diff
@@ -87,6 +87,10 @@ public static NumberConstraint From(BigInteger? min, BigInteger? max)
     public bool CanContain(BigInteger value) =>
         !(value < Min || Max < value);
 
+    public bool Overlaps(NumberConstraint other) =>
+        (Min is null || other.Max is null || Min <= other.Max)
+        && (Max is null || other.Min is null || Max >= other.Min);
```

Comment:
I don't see how I can increase coverage any further due to the fact that you cannot create an unlimited constraint (`NumberConstraint.From(null, null)` returns `null`).

Reply:
I see.
Should this still be a case to test thought? Having other being `null`?

Reply:
No. The method is (purposefully) not null-safe for `other`.

Reply:
Yes - but it's still not clear what happens when you give null.
IMO it's good to add a UT for the sake of documenting the expected behavior.
Unless there's no way that null can ever be given?